fixbackend/notification/email/one_time_email.py

```diff
@@ -0,0 +1,239 @@
+from fixbackend.notification.email.email_messages import EmailMessage
+from fixbackend.notification.notification_service import NotificationService
+from fixbackend.sqlalechemy_extensions import UTCDateTime
+from fixbackend.types import AsyncSessionMaker
+from fixbackend.utils import uid
+
+log = logging.getLogger(__name__)
+
+
+class OneTimeEmailKind(StrEnum):
+    TrialEndNotification = "trial_end"
+
+
+@frozen
+class OneTimeEmail:
+    id: OneTimeEmailId
+    kind: OneTimeEmailKind
+    user_id: Optional[UserId]
+    workspace_id: Optional[WorkspaceId]
+    subject: str
+    text_content: str
+    html_content: Optional[str]
+    scheduled_at: datetime
+    sent_at: Optional[datetime]
+
+
+class OneTimeEmailEntity(Base):
+    __tablename__ = "fire_and_forget_email"
+    id: Mapped[OneTimeEmailId] = mapped_column(GUID, primary_key=True)
+    kind: Mapped[str] = mapped_column(String(length=256), nullable=False)
+    user_id: Mapped[UserId] = mapped_column(GUID, nullable=True, index=True)
+    workspace_id: Mapped[WorkspaceId] = mapped_column(GUID, nullable=True, index=True)
+    subject: Mapped[str] = mapped_column(String(length=1024), nullable=False)
+    text_content: Mapped[str] = mapped_column(Text, nullable=False)
+    html_content: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
+    scheduled_at: Mapped[datetime] = mapped_column(UTCDateTime, nullable=False, index=True)
+    sent_at: Mapped[Optional[datetime]] = mapped_column(UTCDateTime, nullable=True, index=True)
+
+    def to_model(self) -> OneTimeEmail:
+        return OneTimeEmail(
+            id=self.id,
+            kind=OneTimeEmailKind(self.kind),
+            user_id=self.user_id,
+            workspace_id=self.workspace_id,
+            subject=self.subject,
+            text_content=self.text_content,
+            html_content=self.html_content,
+            scheduled_at=self.scheduled_at,
+            sent_at=self.sent_at,
+        )
+
+    @staticmethod
+    def from_model(model: OneTimeEmail) -> "OneTimeEmailEntity":
+
+        return OneTimeEmailEntity(
+            id=model.id,
+            kind=model.kind.value,
+            user_id=model.user_id,
+            workspace_id=model.workspace_id,
+            subject=model.subject,
+            text_content=model.text_content,
+            html_content=model.html_content,
+            scheduled_at=model.scheduled_at,
+            sent_at=model.sent_at,
+        )
+
+
+class OneTimeEmailService(Service):
+    def __init__(
+        self,
+        notification_service: NotificationService,
+        user_repository: UserRepository,
+        session_maker: AsyncSessionMaker,
+        dispatching: bool,
+    ) -> None:
+        self.session_maker = session_maker
+        self.notification_service = notification_service
+        self.user_repository = user_repository
+        self._send_pending_emails: Optional[Periodic] = None
+        self._cleanup_old_emails: Optional[Periodic] = None
```

Comment:
The whole service should be wired only in dispatching. The flag basically turns it on/off.